src/plugins/binmap/binmap-format.js

```diff
@@ -0,0 +1,63 @@
+/// <reference types="@mapeditor/tiled-api" />
+
+/*
+ * binmap-format.js
+ *
+ * This extension adds the 'binary map format' type to the Export As menu.
+ * This is an alternative to the .so plugin, written in JavaScript.
+ * To install, copy to `~/.config/tiled/extensions`
+ *
+ * Simplest format possible. Consist of a 16 bytes long header and then data.
+ * See FORMAT.md for details.
+ */
+
+tiled.registerMapFormat("binmap", {
+	name: "Binary map files",
+	extension: "map",
+
+	write: (map, fileName) => {
+		let numLayers = 0, gid = 1;
+		/* fix missing firstgid properties on tilesets */
+		for (let i = 0; i < map.tilesets.length; ++i) {
+			map.tilesets[i].firstgid = gid;
+			gid += map.tilesets[i].tileCount;
```

Comment:
I'll have to note that using `tileCount` unfortunately only works when there are no gaps in the tile IDs. Image collection tilesets can contain gaps, since when you remove tiles from such a tileset the other tiles don't change their ID.

If you want to support image collection tilesets, you'd currently need to iterate all tiles to find the maximum used tile ID.